packages/ui-library/src/components/virtualized-table/index.tsx

```diff
@@ -86,7 +92,15 @@ function buildVirtuosoTableComponents<D, C extends BaseContext>(): TableComponen
     }),
     Table: (props) => <Table sx={{ padding: 0 }} {...props} />,
     TableHead,
-    TableRow: ({ item: _item, ...props }) => <TableRow {...props} />,
+    TableRow: ({ item: _item, context, ...props }) => (
+      <TableRow
+        {...props}
+        onMouseEnter={() => context?.rowContext?.setHovered && context.rowContext.setHovered(props['data-index'])}
```

Comment:
What do we need this for? and where are we sending data-index?

Reply:
As the hover and click events should be applied to the whole row, I didn't find any way to do this in the consumer component (We only send the children of TableRow for the table body). The only way I found was to apply this props when building the table